modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFolderItemLocalServiceTest.java

```diff
@@ -0,0 +1,353 @@
+
+	@Before
+	public void setUp() throws Exception {
+		_objectFolderA = _addObjectFolder();
+		_objectFolderB = _addObjectFolder();
+		_objectFolderC = _addObjectFolder();
+
+		_objectDefinition1 = ObjectDefinitionTestUtil.addCustomObjectDefinition(
+			_objectFolderA.getObjectFolderId(), _objectDefinitionLocalService);
+		_objectDefinition2 = ObjectDefinitionTestUtil.addCustomObjectDefinition(
+			_objectFolderB.getObjectFolderId(), _objectDefinitionLocalService);
+		_objectDefinition3 = ObjectDefinitionTestUtil.addCustomObjectDefinition(
+			_objectFolderA.getObjectFolderId(), _objectDefinitionLocalService);
+		_objectDefinition4 = ObjectDefinitionTestUtil.addCustomObjectDefinition(
+			_objectFolderC.getObjectFolderId(), _objectDefinitionLocalService);
+
+		ObjectRelationshipTestUtil.addObjectRelationship(
+			_objectRelationshipLocalService, _objectDefinition1,
+			_objectDefinition2);
+		ObjectRelationshipTestUtil.addObjectRelationship(
+			_objectRelationshipLocalService, _objectDefinition1,
+			_objectDefinition4);
+		ObjectRelationshipTestUtil.addObjectRelationship(
+			_objectRelationshipLocalService, _objectDefinition2,
+			_objectDefinition3);
+	}
+
+	@Test
+	public void testAddObjectFolderItem() throws Exception {
+		long objectFolderId = RandomTestUtil.randomLong();
+
+		AssertUtils.assertFailure(
+			NoSuchObjectFolderException.class,
+			"No ObjectFolder exists with the primary key " + objectFolderId,
+			() -> _objectFolderItemLocalService.addObjectFolderItem(
+				TestPropsValues.getUserId(),
+				_objectDefinition1.getObjectDefinitionId(), objectFolderId, 0,
+				0));
+
+		long objectDefinitionId = RandomTestUtil.randomLong();
+
+		AssertUtils.assertFailure(
+			NoSuchObjectDefinitionException.class,
+			"No ObjectDefinition exists with the primary key " +
+				objectDefinitionId,
+			() -> _objectFolderItemLocalService.addObjectFolderItem(
+				TestPropsValues.getUserId(), objectDefinitionId,
+				_objectFolderA.getObjectFolderId(), 0, 0));
+
+		ObjectFolderItem objectFolderItem =
+			_objectFolderItemLocalService.addObjectFolderItem(
+				TestPropsValues.getUserId(),
+				_objectDefinition4.getObjectDefinitionId(),
+				_objectFolderB.getObjectFolderId(), 0, 0);
+
+		Assert.assertEquals(
+			TestPropsValues.getCompanyId(), objectFolderItem.getCompanyId());
+		Assert.assertEquals(
+			TestPropsValues.getUserId(), objectFolderItem.getUserId());
+		Assert.assertEquals(
+			_objectDefinition4.getObjectDefinitionId(),
+			objectFolderItem.getObjectDefinitionId());
+		Assert.assertEquals(
+			_objectFolderB.getObjectFolderId(),
+			objectFolderItem.getObjectFolderId());
+		Assert.assertEquals(0, objectFolderItem.getPositionX());
+		Assert.assertEquals(0, objectFolderItem.getPositionY());
+	}
+
+	@Test
+	public void testAddObjectFolderItemWhenAddingObjectDefinition()
+		throws Exception {
+
+		ObjectFolder objectFolder = _addObjectFolder();
+
+		_assertObjectFolderItems(new Long[0], objectFolder.getObjectFolderId());
```

Comment:
@seltong for me is more easy to understand the test creating new entities.

Reply:
I will need some help from @marcelabc in this case because I'm in a doubt if it would be better reuse the code or not.

if we didn't reuse the code, I believe that would be interesting add an assert after create the `ObjectFolder` to ensure that the folder is empty and to make it clear to the person that is reading the test.